content/docs/rfcs/23/README.md

```diff
@@ -0,0 +1,155 @@
+4) that the data in PubSub field is protobuf (unlike Eth2 where it is `ssz_snappy`) as determined by WakuMessage.
+
+The default PubSub topic SHOULD be used for all protocols.
+However, in certain situations other topics MAY be used.
+
+### Pubsub topic format
+
+PubSub topics SHOULD follow the folloing structure:
+
+`/waku/2/TopicName/Encoding`
+
+This namespaced structure makes things like compatibility and discoverability and automatic handling of new topics easier.
+For example, if the encoding of the payload is changed, compression is introduced, etc.
+
+For more on this format of PubSub topics, see [Ethereum 2 P2P spec](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/phase0/p2p-interface.md#topics-and-messages) where inspiration for this format was taken from.
+
+### Separation of two applications example
+
+Let's say we have some M2M topic that is heavy and experience a lot of heavy traffic and a Status Network one. This can be segregated into:
+
+```
+/waku/2/waku-status/proto
+/waku/2/waku-walletconnect/proto
+```
+
+This indicates that they are WakuMessages but for different domains completely.
+
+### Topic sharding example
+
+Topic sharding is currently not supported by default, but is planned for the future in order to deal with increased network traffic. Here's a sketch for what this might look like:
+
+```
+waku/2/waku-9_shard-0/proto
+...
+waku/2/waku-9_shard-9/proto
+```
+
+This indicates explicitly that the network traffic has been partitioned into 10 buckets.
+
+### Compression example
+
+Not yet implemented, but would be easy to add with:
+
+`/waku/2/default-waku/proto_snappy`
+
+### Why use a single PubSub topic by default?
+
+Using a single PubSub topic ensures a connected network, as well some degree of metadata protection. See [section on Anonymity/Unlinkability](/spec/10/#anonymity--unlinkability).
+
+If you use another PubSub topic, be aware that metadata protection might be weakened,
+and other nodes in the network might not subscribe or store messages for your given Pubsub topic.
+This means you are likely to have to run your own full nodes which may make your application less robust.
+
+## Content topics
+
+The other type of topic that exists in Waku v2 is content topics.
+This is used for content based filtering.
+See [14/WAKU2-MESSAGE spec](/spec/14) for where this is specified.
+Note that this doesn't impact routing of messages between relaying nodes,
+but it does impact how request/reply protocols such as 
+[12/WAKU2-FILTER](https://rfc.vac.dev/spec/12/) and [13/WAKU2-STORE](https://rfc.vac.dev/spec/13/) are used.
+
+This is especially useful for nodes that have limited bandwidth,
+and only want to pull down messages that match this filter.
+
+Since all messages are relayed regardless of content topic, you MAY use any content topic you wish without impacting how messages are relayed.
+
+### Content topic format
+
+The format of content topics is as follows:
+
+`/waku/2/ContentTopic/Encoding`
```

Comment:
Regarding the format of content topic,

First, `/waku/2/`, is it not redundant? given that I am using already the `/waku/2/` topic, do I need to specify as well in the content topic that I am using waku-2? or should it not be given?

Next,
I don't know exactly how it's going to be used so I might be misunderstanding something,
but it seems to me that the last element (i.e `/proto` or `/rlp`) might be also redundant or possibly create some clashes.  

I have partial understanding, so some of the assumptions might be wrong and invalidate some of the scenarios:

I am not 100% sure what `/proto` refers to, so I'll add both cases that I could think of:

### `/proto` refers to `WakuMessage`

So `/proto` means that we are exchanging `WakuMessage`, in which case, is it not redundant as it's already specified at the topic level (the last `/proto`)?

### `/proto` refers to my custom application level protobuf 

Say we are in the old status days, we would use instead `/transit`, just to make it simpler to think about it, if that's the case though, it would not make my application able to speak with any other transit application (or protobuf), so I'd be sort of "squatting" on this very generic content topic, but another developer who uses transit might potentially receive my messages creating clashes.

Overall it seems like as an application developer, I'd rather have more fields to customize, something like:

`/status-chat-application/1/ContentTopic/proto`

so a format like:

`/{application-name}/{version-of-the-application}/{content-topic}/{encoding}`

It's also true that I can stick the applicationID in the content topic to avoid clashes, but looks like less natural:

`/waku/2/{application-name}-{content-topic}/{encoding}`

In any case, it's probably worth explaining or being a bit more prescriptive on how multiple applications are supposed to use content-topics, as we encourage everyone to use the same `topic`, then we might want to be a bit prescriptive on how to use content topic to avoid clashes and/or mentioning that clashes between applications are possible.

Sorry if I misunderstand something, I had a look at the docs and try to cover all the cases I could think of, but I might be completely off point

Reply:
This makes a lot of sense, @cammellos . I especially like the `/{application-name}/{version-of-the-application}/{content-topic}/{encoding}`  format for content.

It also works well for our existing paradigms. For example, if we (roughly) define a Waku v2 application as "anything that defines the `payload` of a WakuMessage", the v1 <> v2 bridge would also be considered an application. In this case the content topic would be `/waku/1/{v1-topic}/rlp`. This clears up the terminology and makes it explicit that `/waku/1` is an application running over `/waku/2`.

`{encoding}` for a content topic would then be the serialization/encoding scheme of the [`WakuMessage.payload`](https://rfc.vac.dev/spec/14/#payloads), while
`{encoding}` for a PubSub topic would be the serialization/encoding scheme of the PubSub [`RPC.Message.data`](https://rfc.vac.dev/spec/11/#protobuf-definitions).

Reply:
Also please the casing that you expect the content topic to use (e.g. kebab case).

Reply:
Great feedback, thanks @cammellos. Agree @jm-clius 

Addressed